hawkular-alerts-engine/src/main/java/org/hawkular/alerts/engine/impl/AlertsEngineCache.java

```diff
@@ -52,12 +52,13 @@ public AlertsEngineCache() {
     /**
      * Check if a specific dataId is active on this node
      *
+     * @param tenantId to check if has triggers deployed on this node
      * @param dataId to check if it has triggers deployed on this node
      * @return true if it is active
      *         false otherwise
      */
-    public boolean isDataIdActive(String dataId) {
-        return dataId != null && activeDataIds.contains(dataId);
+    public boolean isDataIdActive(String tenantId, String dataId) {
+        return tenantId != null && dataId != null && activeDataIds.contains(new DataId(tenantId, dataId));
```

Comment:
Was adding the tenant fixing a bug or added as some sort of formality?  If I recall, ignoring tenant was a conscious decision to avoid the extra object creation and to consolidate the cache entries. Was there an issue with remove being too aggressive?